### PR TITLE
Adds express 4 support

### LINKED ISSUE #738: Upgrade to Express 4, support NodeJS 0.12.6
In order to support NodeJS 0.12 we need to upgrade the version of Express to version 4.  This has breaking changes, which means there's a bit of work involved.

### PR DESCRIPTION
**Pull request by darylhedley**
Fixes #738

### CODE REVIEW

**Comment by darylhedley** [comment]
For testing this you will need to delete all your node_modules in the root and then run npm install again. Also removed the warnings about which node version is usable.

**Comment by dennis-learningpool** [comment]
sweet! testing it now.

**Comment by dennis-learningpool** [comment]
+1 Tested all of the functionality I could think of. Works a treat. Well done!

**Comment by dancgray** [comment]
+1, not tested all functionality, but tried in a few different versions of node without problems.

**Comment by darylhedley** [comment]
Before merging this - we should have some upgrade notes

**Comment by darylhedley** [comment]
Upgrade notes are now with @chucklorenz

File: lib/application.js [hunk 1]
```diff
@@ -259,75 +286,85 @@ Origin.prototype.createServer = function (options, cb) {
...
-      });
+    server.use(compression());
+    /*server.use(favicon());*/
+    server.use(morgan('dev'));
+    server.use(bodyParser.urlencoded({
+      extended: true
+    }));
+    server.use(bodyParser.json());
+    server.use(methodOverride());
+    //server.use(multer({putSingleFilesInArray: true}));
```
Comment (dennis-learningpool) [inline lib/application.js]: @darylhedley cough

**Pull request merged**